buildenv/jenkins/common/variables-functions.groovy

```diff
@@ -730,10 +767,18 @@ def set_artifactory_config(id="Nightly") {
 
     if (VARIABLES.artifactory.defaultGeo) {
         // Allow default geo to be overridden with a param. Used by the Clenaup script to target a specific server.
-        ARTIFACTORY_CONFIG['defaultGeo'] = (params.ARTIFACTORY_GEO) ? params.ARTIFACTORY_GEO : VARIABLES.artifactory.defaultGeo
+        ARTIFACTORY_CONFIG['defaultGeo'] = params.ARTIFACTORY_GEO ?: VARIABLES.artifactory.defaultGeo
         ARTIFACTORY_CONFIG['geos'] = VARIABLES.artifactory.server.keySet()
-        ARTIFACTORY_CONFIG['repo'] = VARIABLES.artifactory.repo
-        ARTIFACTORY_CONFIG['uploadDir'] = "${ARTIFACTORY_CONFIG['repo']}/${JOB_NAME}/${BUILD_ID}/"
+        ARTIFACTORY_CONFIG['repo'] = get_value(VARIABLES.artifactory.repo, id) ?: get_value(VARIABLES.artifactory.repo, 'default')
+        ARTIFACTORY_CONFIG['uploadDir'] = get_value(VARIABLES.artifactory.uploadDir, id) ?: "${ARTIFACTORY_CONFIG['repo']}/${JOB_NAME}/${BUILD_ID}/"
+
+        // If uploadDir has unresolved variables at this point, use Groovy Template Binding engine to resolve them
+        if (ARTIFACTORY_CONFIG['uploadDir'].contains('$')) {
```

Comment:
Instead of seting `ARTIFACTORY_CONFIG['uploadDir']` at L773 then updating at L 780, you could simplify by defining a local variable for `uploadDir`, resolve all placeholders then set `ARTIFACTORY_CONFIG['uploadDir']`

Reply:
Would that be more simple? It just adds another variable we only need temporarily. I did the same thing in #10457 with the SDK_FILENAME